src/main/java/org/assertj/vavr/api/AbstractEitherAssert.java

```diff
@@ -153,6 +154,71 @@ public SELF containsLeftInstanceOf(Class<?> clazz) {
         return myself;
     }
 
+    /**
+     * Verifies that the actual {@link io.vavr.control.Either} contains a right-sided value and gives this value to the given
+     * {@link java.util.function.Consumer} for further assertions. Should be used as a way of deeper asserting on the
+     * containing object, as further requirement(s) for the value.
+     * <p>
+     * Assertions will pass :
+     * <pre><code class='java'> // one requirement
+     * assertThat(Either.right("something")).hasValueSatisfying(it -&gt; assertThat(it).isEqualTo("something"));
+     *
+     * // multiple requirements
+     * assertThat(Either.right("something")).hasValueSatisfying(it -&gt; {
+     *   assertThat(it).isEqualTo("something");
+     *   assertThat(it).startsWith("some");
+     *   assertThat(it).endsWith("thing");
+     * }); </code></pre>
+     *
+     * Assertions will fail :
+     * <pre><code class='java'>
+     * assertThat(Either.right("something")).hasValueSatisfying(it -&gt; assertThat(it).isEqualTo("something else"));
+     *
+     * // fail because Either is left-sided, there is no value to perform assertion on
+     * assertThat(Either.left(42)).hasValueSatisfying(it -&gt; {});</code></pre>
+     *
+     * @param requirement to further assert on the right-sided object contained inside the {@link io.vavr.control.Either}.
+     * @return this assertion object.
+     */
+    public SELF hasValueSatisfying(Consumer<RIGHT> requirement) {
```

Comment:
Note I specifically didn't call this `hasRightValueSatisfying`. I think `hasValueSatisfying` is more consistent with the terminology used in `Either` as `get()` is not called `getRight`. And it keeps it consistent with the other `hasValueSatisfying` methods in assertj.
I realize this makes it kind of inconsistent in this class so I'm cool if you want me to rename it.

Reply:
Your reasoning is valid and I see we should go with this naming convention from the beginning. However, I would opt for naming it as `hasRightValueSatisfying` to have the API consistent at the moment. In the future major release, we can change the API properly.